hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightEndpoint.java

```diff
@@ -674,6 +686,625 @@ public Response getDeletedDirectorySummary() {
+    OmBucketInfo bucketInfo =
+        omMetadataManager.getBucketTable().getSkipCache(bucketKey);
+    long bucketId = bucketInfo.getObjectID();
+    if (names.length == 2) {
+      return ReconUtils.constructObjectPathWithPrefix(volumeId, bucketId);
+    }
+
+    // Fetch the immediate parentID which could be a directory or the bucket itself
+    BucketHandler handler =
+        getBucketHandler(reconNamespaceSummaryManager, omMetadataManager,
+            reconSCM, bucketInfo);
+    long dirObjectId = -1;
+    try {
+      OmDirectoryInfo dirInfo = handler.getDirInfo(names);
+      if (null != dirInfo) {
+        dirObjectId = dirInfo.getObjectID();
+      }
+    } catch (Exception ioe) {
+      LOG.error("Not valid directory :{}", ioe);
+    }
+    return ReconUtils.constructObjectPathWithPrefix(volumeId, bucketId, dirObjectId);
+  }
+
+  /**
+   * Common method to retrieve keys from a table based on a search prefix and a limit.
+   *
+   * @param table     The table to retrieve keys from.
+   * @param paramInfo The stats object holds total count, count and limit.
+   * @return A map of keys and their corresponding OmKeyInfo objects.
+   * @throws IOException If there are problems accessing the table.
+   */
+  private Map<String, OmKeyInfo> retrieveKeysFromTable(
+      Table<String, OmKeyInfo> table, ParamInfo paramInfo)
+      throws IOException {
+    boolean skipPrevKey = false;
+    String seekKey = paramInfo.getPrevKey();
+    Map<String, OmKeyInfo> matchedKeys = new LinkedHashMap<>();
+    try (
+        TableIterator<String, ? extends Table.KeyValue<String, OmKeyInfo>> keyIter = table.iterator()) {
+
+      if (!paramInfo.isSkipPrevKeyDone() && StringUtils.isNotBlank(seekKey)) {
+        skipPrevKey = true;
+        Table.KeyValue<String, OmKeyInfo> seekKeyValue =
+            keyIter.seek(seekKey);
+
+        // check if RocksDB was able to seek correctly to the given key prefix
+        // if not, then return empty result
+        // In case of an empty prevKeyPrefix, all the keys are returned
+        if (seekKeyValue == null || (!seekKeyValue.getKey().equals(paramInfo.getPrevKey()))) {
+          return matchedKeys;
+        }
+      } else {
+        keyIter.seek(paramInfo.getStartPrefix());
+      }
+
+      while (keyIter.hasNext()) {
+        Table.KeyValue<String, OmKeyInfo> entry = keyIter.next();
+        String dbKey = entry.getKey();
+        if (!dbKey.startsWith(paramInfo.getStartPrefix())) {
+          break; // Exit the loop if the key no longer matches the prefix
+        }
+        if (skipPrevKey && dbKey.equals(paramInfo.getPrevKey())) {
+          paramInfo.setSkipPrevKeyDone(true);
+          continue;
+        }
+        if (applyFilters(entry, paramInfo)) {
+          if (matchedKeys.size() >= paramInfo.getLimit()) {
+            break;
+          }
+          matchedKeys.put(dbKey, entry.getValue());
+          paramInfo.setLastKey(dbKey);
```

Comment:
we should break as soon as it reach limit, after matchedKeys.put()

Reply:
Ok done.